Python/Product/Analysis/Intellisense/LineInfo.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+using Microsoft.PythonTools.Parsing;
+
+namespace Microsoft.PythonTools.Intellisense {
+    class LineInfo {
+        public int Start, Length;
+        /// <summary>
+        /// 0-based line number
+        /// </summary>
+        public int LineNo;
+        public NewLineKind LineBreak;
```

Comment:
Why these fields aren't read-only?

Reply:
Also, I believe this type can be a struct.

Reply:
No idea, I didn't write this class. But you're right, it does only seem to be treated as an integral value, so it can be a struct.